homeassistant/helpers/intent.py

```diff
@@ -305,16 +364,29 @@ def async_set_card(
         """Set card response."""
         self.card[card_type] = {"title": title, "content": content}
 
+    @callback
+    def async_set_error(self, reason: IntentResponseErrorReason, message: str) -> None:
+        """Set response error."""
+        self.state = IntentResponseState.FAILURE
```

Comment:
do we still need this if we can already distinguish from `response_type` if it's a failure?